src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,205 @@
+                    col_value = $"CAST({col_value} AS NVARCHAR(MAX))";
+                }
+
+                // Create json. Example: "book.title": "Title" would be a sample output.
+                stringAgg += $"\"{column.Label}\":\"\' + STRING_ESCAPE(ISNULL({col_value},''),'json') + \'\"";
+                i++;
+
+                // Add comma if not last column. example: {"id":"1234","name":"Big Company"}
+                // the below ensures there is a comman after id but not after name.
+                if (i != structure.Columns.Count)
+                {
+                    stringAgg += ",";
+                }
+            }
+
+            if (structure.IsListQuery)
+            {
+                // Array wrappers if we are trying to get a list of objects.
+                columns = $"COALESCE(\'[\'+STRING_AGG(\'{{{stringAgg}}}\',', ')+\']\',\'[]\')";
+            }
+            else
+            {
+                columns = $"STRING_AGG(\'{{{stringAgg}}}\',', ')";
+            }
+
+            // the fromSql generates the query that will return the actual results that will be jsonified
+            // in format of earlier code.
+            string fromSql = $"{BuildSqlQuery(structure)}";
+            string query = $"SELECT {columns}"
+                + $" FROM ({fromSql}) AS {QuoteIdentifier(structure.SourceAlias)}";
+            return query;
+
+        }
+
+        /// <summary>
+        /// Build internal query for DW.
+        /// This will generate the query that will return results in sql.
+        /// Results of this query will be jsonified and returned to the user.
+        /// </summary>
+        public string BuildSqlQuery(SqlQueryStructure structure)
+        {
+            string dataIdent = QuoteIdentifier(SqlQueryStructure.DATA_IDENT);
+            string fromSql = $"{QuoteIdentifier(structure.DatabaseObject.SchemaName)}.{QuoteIdentifier(structure.DatabaseObject.Name)} " +
+                             $"AS {QuoteIdentifier($"{structure.SourceAlias}")}{Build(structure.Joins)}";
+
+            fromSql += string.Join(
+                    "",
+                    structure.JoinQueries.Select(
+                        x => $" OUTER APPLY ({Build(x.Value)}) AS {QuoteIdentifier(x.Key)}({dataIdent})"));
+            string predicates = JoinPredicateStrings(
+                                    structure.GetDbPolicyForOperation(EntityActionOperation.Read),
+                                    structure.FilterPredicates,
+                                    Build(structure.Predicates),
+                                    Build(structure.PaginationMetadata.PaginationPredicate));
+            string columns = WrappedColumns(structure);
+            string orderBy = $" ORDER BY {Build(structure.OrderByColumns)}";
+
+            string query = $"SELECT TOP {structure.Limit()} {columns}"
+                + $" FROM {fromSql}"
+                + $" WHERE {predicates}"
+                + orderBy;
+            return query;
```

Comment:
Can we add the sample content of query as code comment for better understanding?

Reply:
fixed.